### PR TITLE
CA-207783: Honour additional options passed to NFS SR creation

### PR DESCRIPTION
**Pull request by chandrikas**
Signed-off-by: Chandrika Srinivasan <chandrika.srinivasan@citrix.com>

### CODE REVIEW

File: drivers/ISOSR.py [hunk 1]
```diff
@@ -242,13 +242,14 @@ def attach(self, sr_uuid):
 
         self.credentials = os.path.join("/tmp", util.gen_uuid())
         if self.dconf.has_key('type'):
+            # Creation via XC
```
Comment (germanop) [inline drivers/ISOSR.py]: Why is only XC specific?

Comment (chandrikas) [inline drivers/ISOSR.py]: That's how we receive it from XAPI. There is an additional type dconf when the request comes in from XC

File: drivers/NFSSR.py [hunk 2]
```diff
@@ -92,6 +92,10 @@ def load(self, sr_uuid):
         if self.dconf.has_key('useUDP') and self.dconf['useUDP'] == 'true':
             self.transport = "udp"
         self.nfsversion = nfs.validate_nfsversion(self.dconf.get('nfsversion'))
+        if self.dconf.has_key('options'):
```
Comment (germanop) [inline drivers/NFSSR.py]: Not an error and I do understand it is to adhere to code above but
has_key() is deprecated.

Sooner or later we should start "breaking style compatibility" and write standard compliant code.

Up to you

Comment (chandrikas) [inline drivers/NFSSR.py]: Good point. I'll change it to 
`if 'options' in self.dconf`

**Comment by germanop** [comment]
Acked-by: Germano Percossi \<germano.percossi@citrix.com\>

**Pull request closed without merging**